src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -408,24 +427,56 @@ FireStoreDirectoryEntry computeDirectory(
         String fullPath = fireStoreUtils.getFullPath(dirEntry.getPath(), dirEntry.getName());
         List<FireStoreDirectoryEntry> enumDir = directoryDao.enumerateDirectory(firestore, snapshotId, fullPath);
 
-        // Recurse to compute results from underlying directories
         List<FireStoreDirectoryEntry> enumComputed = new ArrayList<>();
-        for (FireStoreDirectoryEntry dirItem : enumDir) {
-            if (dirItem.getIsFileRef()) {
-                // Read the file metadata to get the size and checksum. We do a bit of a hack and copy
-                // the size and checksums into the in-memory dirItem. That way we only compute on the directory
-                // objects.
-                FireStoreFile file =
-                    fileDao.retrieveFileMetadata(firestore, dirItem.getDatasetId(), dirItem.getFileId());
-
-                dirItem
-                    .size(file.getSize())
-                    .checksumMd5(file.getChecksumMd5())
-                    .checksumCrc32c(file.getChecksumCrc32c());
-
-                enumComputed.add(dirItem);
-            } else {
-                enumComputed.add(computeDirectory(firestore, snapshotId, dirItem, updateBatch));
+
+        // Recurse to compute results from underlying directories
+        try (Stream<FireStoreDirectoryEntry> stream = enumDir.stream()) {
+            enumComputed.addAll(stream
+                .filter(f -> !f.getIsFileRef())
+                .map(f -> {
+                    try {
+                        return computeDirectory(firestore, snapshotId, f, updateBatch);
+                    } catch (InterruptedException e) {
+                        throw new ApiException("Error computing directory metadata", e);
```

Comment:
We generally use specific exceptions that derive from base class exceptions in common/exceptions. Those exceptions are associated with Http status results as well. For example, `bio/terra/service/filedata/exception/FileNotFoundException.java`

Reply:
ok, good to know

Reply:
I guess in this case for an interrupt, I wasn't sure what common exception we would have for that so ApiException looked reasonable

Reply:
I think this should come back to the client as a 500, so i would make an exception that derived from InternalServerErrorException.

Reply:
👍